src/Collection/Collection.service.ts

```diff
@@ -131,4 +151,70 @@ export class CollectionService {
 
     return true
   }
+
+  public async deleteCollection(collectionId: string): Promise<void> {
+    const collection = await this.getDBCollection(collectionId)
+    if (collection.urn_suffix !== null && collection.third_party_id !== null) {
+      // If it's a TPC we must check if there's an item already published under that collection urn suffix
+      const collectionItems = await thirdPartyAPI.fetchThirdPartyCollectionItems(
+        collection.third_party_id,
+        collection.urn_suffix
+      )
+      if (collectionItems.length > 0) {
+        throw new CollectionAlreadyPublishedException(
+          collection.id,
+          CollectionType.THIRD_PARTY,
+          'deleted'
+        )
+      }
+    } else {
+      // If it's a DCL collection, we must check if it was already published
+      if (await this.isPublished(collection.contract_address!)) {
+        throw new CollectionAlreadyPublishedException(
+          collectionId,
+          CollectionType.DCL,
+          'deleted'
+        )
+      }
+    }
+    if (this.isLockActive(collection.lock)) {
+      throw new CollectionLockedException(collection.id, 'deleted')
+    }
+    await Promise.all([
+      Collection.delete({ id: collection.id }),
+      // TODO: This should eventually be in the item's service
+      Item.delete({ collection_id: collection.id }),
+    ])
+  }
+
+  public async isOwnedOrManagedBy(
+    id: string,
+    ethAddress: string
+  ): Promise<boolean> {
+    const collection = await Collection.findOne<CollectionAttributes>(id)
+    if (collection && collection.third_party_id && collection.urn_suffix) {
+      return isManager(
+        getThirdPartyCollectionURN(
+          collection.third_party_id,
+          collection.urn_suffix
+        ),
+        ethAddress
+      )
+    } else if (collection) {
+      return collection.eth_address === ethAddress
+    }
+
+    return false
+  }
+
+  private async getDBCollection(
```

Comment:
Questions:
- We have a `getDBCollection` on the utils file that just does this: `export const getDBCollection = (id: string) => Collection.findOne<CollectionAttributes>(id)` should that method be updated to throw and used here too?
- should we use this method in the rest of the file?

Reply:
I've just removed the getDBCollection method in the utils file (replacing it in the function with its content) to avoid any confusion. Any operations with the collections should come from the Collections service. The utils file has a couple of things that might be part of the CollectionService or another service IMO.